fenix/app/src/main/java/org/mozilla/fenix/shopping/ReviewQualityCheckFragment.kt

```diff
@@ -66,4 +70,18 @@ class ReviewQualityCheckFragment : BottomSheetDialogFragment() {
             }
         }
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        store.flowScoped(viewLifecycleOwner) { flow ->
+            val initial = Pair<ReviewQualityCheckState?, ReviewQualityCheckState?>(null, null)
+            flow.scan(initial) { acc, value ->
+                Pair(acc.second, value)
+            }.filter {
+                it.first is ReviewQualityCheckState.Initial && it.second is ReviewQualityCheckState.NotOptedIn
+            }.collect {
+                behavior?.state = BottomSheetBehavior.STATE_EXPANDED
```

Comment:
Would it be worth to maybe cancel the scope after successfully expanding the bottom sheet?

Reply:
Not really. Since we want to set this state whenever we move from `Initial` to `NotOptedIn` state and that can occur multiple times if user doesn't opt in and keeps opening the bottom sheet. The flow is scoped to Fragment's [viewLifecycleScope](https://developer.android.com/topic/libraries/architecture/coroutines#lifecyclescope) so it will auto cancel when the fragment is destroyed.

Reply:
Good point, we should cancel `onDestroy`, will look into it!

Reply:
Updated to use `AbstractBinding` which cancels the scope based on lifecycle.